wpilibc/src/main/native/cpp/drive/DifferentialDrive.cpp

```diff
@@ -112,7 +112,7 @@ DifferentialDrive::WheelSpeeds DifferentialDrive::ArcadeDriveIK(
   double maxInput =
       std::copysign(std::max(std::abs(xSpeed), std::abs(zRotation)), xSpeed);
 
-  if (xSpeed >= 0.0) {
```

Comment:
The input space is split into quadrants based on the signedness of xSpeed and zRotation. Why is this change more correct?

Reply:
The unexpected behavior only occurs because -0.0 copies its sign to maxInput making maxInput negative while checking xSpeed >= 0 returns true. This results in a situation where maxInput is negative but the code enters the first if clause instead of the else clause (where it should be if maxInput is negative).

The code only correctly sets leftSpeed and rightSpeed if maxInput is positive and the code enters the if clause, or if maxInput is negative and the code enters the else clause.

By checking xSpeed >= 0, the following 3 situations can occur:
- maxInput is positive and xSpeed is positive or 0.0. This results in the code entering the if clause and setting intended speed values
- maxInput is negative and xSpeed is negative. This results in the code entering the else clause and setting intended speed values
- maxInput is 0.0 or -0.0. For maxInput to be equal to zero, both xSpeed and zRotation must be equal to zero, in which case all code results in the intended speed values of leftPower = 0 and rightPower = 0
- maxInput is negative and xSpeed -0.0. Because xSpeed >= 0 -> -0.0 >= 0 -> true, this results in the code entering the if else and setting unintended speed values


By directly checking maxInput >= 0 instead of xSpeed >= 0, it ensures the final case can never occur, hence solving the issue.